ui/static/ui/js/taxonomy/add_terms_component.jsx

```diff
@@ -7,6 +7,16 @@ define('add_terms_component', ['react', 'lodash', 'jquery', 'uri',
       render: function () {
         var repoSlug = this.props.repoSlug;
         var thiz = this;
+        if (!_.isEmpty(this.props.vocabularies)) {
+          // sort vocabularies
+          this.props.vocabularies.sort(function (vocabLeft, vocabRight) {
```

Comment:
Props must be immutable. Instead you need to fix this in TaxonomyComponent in addTerm (I was wrong when I said AddTermsComponent)